### DISCUSSION TITLE
1.1.409 New model supports

### DISCUSSION THREAD

**Discussion by lllyasviel** [Mikubill/sd-webui-controlnet#2086] [General]
Below new models are supported

    kohya_controllllite_xl_blur_anime_beta.safetensors
    kohya_controllllite_xl_blur_anime.safetensors
    kohya_controllllite_xl_blur.safetensors
    kohya_controllllite_xl_canny.safetensors
    kohya_controllllite_xl_depth.safetensors
    t2i-adapter_diffusers_xl_canny.safetensors
    t2i-adapter_diffusers_xl_depth_midas.safetensors
    t2i-adapter_diffusers_xl_depth_zoe.safetensors
    t2i-adapter_diffusers_xl_lineart.safetensors
    t2i-adapter_diffusers_xl_openpose.safetensors
    t2i-adapter_diffusers_xl_sketch.safetensors

All files are mirrored at https://huggingface.co/lllyasviel/sd_control_collection/tree/main (Use this link for downloading models)

You can download all files in the above link; the list of all original sources are [here (do NOT use this link for downloading models)](https://huggingface.co/lllyasviel/sd_control_collection). Feel free to let us know if you are original authors of some files and want to add/remove files from the list.

You can put models in `stable-diffusion-webui\extensions\sd-webui-controlnet\models` or `stable-diffusion-webui\models\ControlNet`.

# UI change: "blur" preprocessor added to "tile" group

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/8803ab11-37bf-4448-8527-3de65fde2f02)

# About VRAM

All methods have been tested with 8GB VRAM and 6GB VRAM. The 6GB VRAM tests are conducted with GPUs with float16 support.

For 8GB~16GB vram (including 8GB vram), the recommended cmd flag is "--medvram-sdxl". You may edit your "webui-user.bat" as

    @echo off
    
    set PYTHON=
    set GIT=
    set VENV_DIR=
    set COMMANDLINE_ARGS=
    
    call webui.bat --medvram-sdxl --xformers

For vram less than 8GB (like 6GB or 4GB, excluding 8GB vram), the recommended cmd flag is "--lowvram". You may edit your "webui-user.bat" as

    @echo off
    
    set PYTHON=
    set GIT=
    set VENV_DIR=
    set COMMANDLINE_ARGS=
    
    call webui.bat --lowvram --xformers

# About Speed

See the speed collection [here](https://github.com/Mikubill/sd-webui-controlnet/discussions/2061).

**Comment by Michoko92** [comment]
Absolutely awesome! Thank you so much for your amazing work! 🙏

**Comment by Michoko92** [comment]
Hmm, I'm getting pretty strange (and ugly) results with Lineart model.

Here is the guiding image:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/91e59761-b3ab-4d27-a869-9d8ccb0df6d6)

And the result:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/e585aa9f-9f88-4000-af7e-5621782a670a)

My settings:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/172333a4-00f5-497a-a1ee-6aace3a4b585)

Prompt data:

an indigenous little girl sitting against a tree, art by greg olsen
Negative prompt: long neck, disfigured, deformed, cartoon, bad art, poorly drawn, extra limbs, strange, blurry, repetitive, cropped
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 734948626, Size: 768x1152, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, ControlNet 0: "Module: lineart_realistic, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Resize and Fill, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: ControlNet is more important", Version: v1.4.0-RC-1004-g5ef669de

I get consistently ugly results with those settings. Do I do something wrong?

Edit: I installed version 1.1.409 of the extension, of course. :)

**Reply by lllyasviel** [reply]
wrong control mode

**Comment by Michoko92** [comment]
Ok thank you. So you mean only balanced mode is supported? Is it by design? Because the balanced mode seems to follow the input image more loosely, and I know that with SD 1.5, "controlnet is more important" allowed the most precise results. 

I personally use lineart to do render style transfer: for example I have a regular digital art image I'm happy with , then use a strongly styled model ( comic book oriented) to recreate the exact same image in a comic style. Hence the need to follow the lines precisely. Maybe the same workflow is possible now with other SDXL controlnets? 

Hope I make sense 🙂

**Reply by Michoko92** [reply]
Hmm, I think there is a bigger issue? I just tried Balanced mode with lineart model, and here is what I get:

![05259-2091358503-20-7-DPM++ 2M Karras-an indigenous little girl sitting against a tree, art by greg olsen](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/bf110f47-1e06-48e4-9918-0855baf66ac4)

an indigenous little girl sitting against a tree, art by greg olsen
Negative prompt: cartoon, long neck, disfigured, deformed, cartoon, bad art, poorly drawn, extra limbs, strange, blurry, repetitive, cropped
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 2091358503, Size: 768x1152, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, ControlNet 0: "Module: none, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Resize and Fill, Low Vram: False, Guidance Start: 0, Guidance End: 1, Pixel Perfect: True, Control Mode: Balanced", Version: v1.4.0-RC-1004-g5ef669de

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/0f034ac4-4dde-42cb-8b9b-8c913014bc98)

**Reply by lllyasviel** [reply]
hi - may you try tencent official demo at https://huggingface.co/spaces/TencentARC/T2I-Adapter-SDXL
and post some results here?

**Reply by Michoko92** [reply]
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/80ec21f0-30ea-420e-a359-c5b064c4f8eb)

Just tried and got this image with the same input, which is indeed a great result.

**Reply by lllyasviel** [reply]
hi do you have parameters of this image
I will look into this soon

**Reply by Michoko92** [reply]
Prompt:an indigenous little girl sitting against a tree, art by greg olsen
Negative prompt: extra digit, fewer digits, cropped, worst quality, low quality, glitch, deformed, mutated, ugly, disfigured
Style: digital art

Steps: 25
Guidance scale : 5
Adapter conditioning scale: 1
Adapter conditioning factor: 1
Seed: 42

And here's the exact guiding image, just in case: 

![image (1)](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/3c25fe2f-de80-4471-8c46-4eba23f70737)

Thank you! 🙏

**Reply by lllyasviel** [reply]
OK. this extension currently use SAI's comfy behaviors.
If tencent behavior is different, we will give up SAI's comfy behaviors and resort ro tencent behavior because t2i-adapter's official is tencent

**Reply by lllyasviel** [reply]
hello I cannot reproduce the problem
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/f39861b5-ef30-40d8-83bf-6fbf93ae1bd3)
result looks good to me
![00002-42](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/43e5ef4c-c674-4fc1-ba09-43ede929cbf0)
concept art an indigenous little girl sitting against a tree, art by greg olsen . digital artwork, illustrative, painterly, matte painting, highly detailed
Negative prompt: photo, photorealistic, realism, ugly, extra digit, fewer digits, cropped, worst quality, low quality, glitch, deformed, mutated, ugly, disfigured
Steps: 20, Sampler: Euler a, CFG scale: 5, Seed: 42, Size: 896x1152, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, RNG: CPU, ControlNet 0: "Module: none, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Crop and Resize, Low Vram: False, Guidance Start: 0, Guidance End: 1, Pixel Perfect: True, Control Mode: Balanced", SGM noise multiplier: True, Version: v1.6.0

note that "Style: digital art" is the template:

        "prompt": "concept art {prompt} . digital artwork, illustrative, painterly, matte painting, highly detailed",
        "negative_prompt": "photo, photorealistic, realism, ugly"

you may need sdxl style extension to use it.

**Reply by Michoko92** [reply]
Thank you very much for testing. OK so I made further tests, to see what differs between your settings and mine, and things get even weirder!

Indeed, the problem seems to come from the input image (the black and white lineart one). If I use the one generated by the extension, it fails. If I use the one generated by the TancentArt demo, it works fine, like it did for you.

So basically, if you use that one below (generated by the extension), it will fail:

![266785648-91e59761-b3ab-4d27-a869-9d8ccb0df6d6](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/3bb8e828-5f19-443c-8864-f4873edb0f94)

If you use the one I put in my last post (and that you used for your latest test), it will work! ^_^' So it seems that the extension produces a black and white image that the model won't be able to use. I really can't figure out why, but if you use the image above as a guiding input, I'm pretty sure it will fail for you too.

**Reply by lllyasviel** [reply]
ok i will take a look again soon

**Reply by lllyasviel** [reply]
ok I figure out what happened
the tencent model does not support pixel-perfect
you must use 512 as preprocessor resolution like
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/bce15579-e2e3-4cb1-9ec4-1bbc2a5d52f4)

This is a really news for comfyui because comfyui does not even have "preprocessor resolution" and they always use a fixed number by default (as far as i know), but in webui this is at least still possible and can be turned off.

I should probably write at somewhere

**Reply by lllyasviel** [reply]
https://github.com/Mikubill/sd-webui-controlnet/discussions/2099

**Reply by Michoko92** [reply]
Thank you very much for your feedback, @lllyasviel , it's much appreciated. I'd really love to say that unchecking pixel perfect alone, and leaving preprocessor resolution to 512 will always do the trick. Unfortunately, in my case, it definitely doesn't, and I think there are other elements involved.

A quick easily reproducible test:

Input image:
![266858307-80ec21f0-30ea-420e-a359-c5b064c4f8eb](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/8fbf4c73-1612-4895-acef-7b43124a0198)

Settings:

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/be8e52e0-964f-466a-9b12-2cd0cc22d2a5)

an indigenous little girl sitting against a tree, art by greg olsen
Negative prompt: photo, photorealistic, realism, ugly, extra digit, fewer digits, cropped, worst quality, low quality, glitch, deformed, mutated, ugly, disfigured
Steps: 25, Sampler: Euler a, CFG scale: 5, Seed: 42, Size: 896x1152, Model hash: e6bb9ea85b, Model: sd_xl_base_1.0_0.9vae, ControlNet 0: "Module: lineart_realistic, Model: t2i-adapter_diffusers_xl_lineart [bae0efef], Weight: 1, Resize Mode: Resize and Fill, Low Vram: False, Processor Res: 512, Guidance Start: 0, Guidance End: 1, Pixel Perfect: False, Control Mode: Balanced", Version: v1.4.0-RC-1004-g5ef669de

Result : 

![05289-42-25-5-Euler a-an indigenous little girl sitting against a tree, art by greg olsen](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/90d19eec-20b8-45cf-b49c-b2ffbfe416a7)

Result with TencentART demo (same settings, prompt, seed):

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/5762620b-3edd-423a-b907-406b8f4f7708)

I really checked the settings 10 times, so either I'm really dumb (which is not impossible ;)), or there is something else that doesn't work properly.

**Reply by lllyasviel** [reply]
do you compare their preprocessor outputs

**Reply by Michoko92** [reply]
Additional note: If I disable the preprocessor in A1111, and provide the black and white lineart generated by the Tencent demo as input from the test above, then I get good results. So the problem seems definitely linked to the data generated by the preprocessor (but unckecking pixel perfect alone isn't solving it completely)

**Reply by lllyasviel** [reply]
post both preprocessor outputs

**Reply by Michoko92** [reply]
Here's the one by Tencent:

![imagebw](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/a72f19b3-3622-4ef4-ae35-295d8b5ed0c7)

Here's the one by A1111:

![tmpmx4wx34u](https://github.com/Mikubill/sd-webui-controlnet/assets/3066173/cc86b1e5-fd2e-4a50-90ce-9a5343142f0c)

**Reply by lllyasviel** [reply]
what the hell - the model is trained on 384 resolution???????

why they use that low resolution of lineart ???????

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/2951c938-bce3-478f-b136-02a9f1b9ce59)

well you can do the same thing with

![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/25a6b5b7-809c-4b67-9821-cdcbb3a51582)

I really do not understand this decision - well sometimes things like this just happens

I will edit that post again

**Reply by lllyasviel** [reply]
well it works
![image](https://github.com/Mikubill/sd-webui-controlnet/assets/19834515/91944853-9704-4cf2-9182-34a9be23cef2)

you need to use the very low resolution 384 - sometimes i really cannot get the ideas of these people especially when some models are from diffusers

**Reply by Michoko92** [reply]
Thank you very much, it seems indeed that it improves things a lot. I'll make further tests later, and report possible issues, if any. Again, thank you for your great support and work. Cheers!

Edit: would that low resolution explain why those models are so small in size?

**Reply by kasimgu2000** [reply]
I have the exact same issue with either of the Lineart versions. Lowering the preprocessor resolution to 384 doesnt seem to work either....was there any successful repeatable fix to this yet?

**Comment by CCRcmcpe** [comment]
I found controllite models have no effect at all with no errors? But T2I adapters works anyway.

**Reply by CCRcmcpe** [reply]
https://github.com/Mikubill/sd-webui-controlnet/issues/2092 issue raised.

**Comment by ivanoff13** [comment]
please show examples of how it works

**Comment by Jonseed** [comment]
What does the blur_gaussian preprocesser and blur model do? I'm guessing it does something similar to Tile since it is combined in that same group, so sharpens a blurry image? How does it compare to Tile? Are there examples of these somewhere?

**Comment by if-ai** [comment]
̶I̶t̶ ̶i̶s̶ ̶n̶o̶t̶ ̶t̶o̶o̶ ̶o̶b̶v̶i̶o̶u̶s̶ ̶b̶u̶t̶ ̶I̶ ̶w̶a̶n̶t̶ ̶t̶o̶ ̶u̶s̶e̶ ̶a̶ ̶b̶i̶n̶ ̶m̶o̶d̶e̶l̶ ̶I̶ ̶f̶o̶u̶n̶d̶ ̶o̶n̶ ̶t̶h̶e̶ ̶i̶n̶t̶e̶r̶n̶e̶t̶ ̶h̶o̶w̶ ̶I̶ ̶c̶o̶n̶v̶e̶r̶t̶ ̶i̶t̶ ̶t̶o̶ ̶s̶a̶f̶e̶ ̶t̶e̶n̶s̶o̶r̶ ̶o̶r̶ ̶u̶s̶e̶ ̶t̶h̶e̶ ̶b̶i̶n̶ ̶w̶i̶t̶h̶ ̶t̶h̶e̶ ̶C̶N̶ ̶e̶x̶t̶e̶n̶s̶i̶o̶n̶ ̶h̶e̶r̶e̶ ̶i̶s̶ ̶t̶h̶e̶ ̶m̶o̶d̶e̶l̶ ̶i̶n̶ ̶q̶u̶e̶s̶t̶i̶o̶n̶ ̶h̶t̶t̶p̶s̶:̶/̶/̶h̶u̶g̶g̶i̶n̶g̶f̶a̶c̶e̶.̶c̶o̶/̶h̶9̶4̶/̶I̶P̶-̶A̶d̶a̶p̶t̶e̶r̶/̶b̶l̶o̶b̶/̶m̶a̶i̶n̶/̶m̶o̶d̶e̶l̶s̶/̶i̶p̶-̶a̶d̶a̶p̶t̶e̶r̶-̶p̶l̶u̶s̶-̶f̶a̶c̶e̶_̶s̶d̶1̶5̶.̶b̶i̶n̶

no worries I found if I change the extension to pth or safetensors it works
